### ISSUE TITLE
Quenstions about Training details

### ISSUE THREAD

**Issue by QZ1-boy** [Mr-Zwkid/D-FCGS#3]
Thank you for your interesting work and released code.

There are some quenstions about training details when implement code:

In your paper, 3 scenes from Meetroom and 28 scenes from WideRange4D are adopted for training. Which 28 scenes in WideRange4D are adopted?  

These 3+28 scenes are conducted by  python scripts/run_dfcgs_train.py to obtain the respectively ckpt/model.pth. These 3+28 scenes are independently executed by the Python script run_cfcgs_train.py to obtain their respectively ckpt/model.pth? Which  ckpt/model.pth are adopted to conduct inference on testing scenes for generating the final results? 

Are all scenes using the first 300 frames to training except Immersive dataset? 

Except for the immersive dataset (150 frames), do all scenes in the dataset are trained using the first 300 frames? Only scenes of N3DV dataset used x2 donwsampling (--scale_factor 2) ? 

Best regards

**Comment by Mr-Zwkid** [comment]
Sorry for the late response. 

(1) Actually, select 28 scenes from the 'Sunny' subset of WideRange4D dataset, and use the data preprocess scripts to get init and 3DGStream GS. We select the complete sequences with bearable point numbers and size (like < 100MB per ply), as restricted by the GPU memory. 

(2) The checkpoint is obtained using all these training GS sequences as a one-for-all model.

(3) For the commonly used N3V and MeetRoom dataset, the first 300 frames are tested as a common practice in other related paper (like 4DGC and QUEEN).

(4) Well, the model trained from WideRange4D and Parts of MeetRoom Dataset is used for infering on other datasets. The training and inference are separate, as D-FCGS is a **feedforward** codec instead of optimization-based ones. Plus,  you can use x2 donwsampling for all scenes.

If you still have confusions, please let me know. Also, feel free to star this repo😁

**Comment by QZ1-boy** [comment]
Thank you very much for your kindly response!

I have few questions about orginazation of training dataset and detailed training procedure. 

How to achieve the training of all these training GS sequences once to obtain a one-for-all model? 
In my understand, two training procedure may be proposed, i.e., scene-by-scene training or intergate all these training GS sequences as a scene. If using scene-by-scene training, the trained model of scene-1 as the pre-trained model to initialize the scene-2? If using intergate all these training GS sequences as a scene,  so the intergated scene has the 28x60 camera views, but these views may be unrelated due to the comes from different scenarios.  

Could you tell me the specifical training procedure and how to intergate these training GS sequences?

Best regards

**Comment by Mr-Zwkid** [comment]
Well, we should clarify that the data preprocess and training are two different procedures.

- data preprocess using `derive_gs_frames.py`: aim to prepare GS point cloud sequences for later training.
- training using `train_dfcgs.py`: aim to train a unified compression model for GS sequences. In this process, each time we utilize two GS ply to get motion tensors to train the compession model using render loss and size loss, which is scene-agnostic and the GS ply remain unchanged (just as image/video for image/video codec training).

You can use the `run_dfcgs_train.py` and enrich the `dataset_list` and `scene_list`  to use these derived GS sequences for training.

**Comment by QZ1-boy** [comment]
I got it！The scene-agnostic design is a wonderful method. Thank you for your patient. 

Best regards

**Comment by Mr-Zwkid** [comment]
You're welcome! I'll close this issue and feel free to reopen it or open another issue if you still have problems.
